packages/components/core/config/format.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @param {Number|String} value 字符串或着数字
+ * @param {Number} precision 小数部分的精度
+ * @returns {object} 返回对象具有value、int、decimal三个属性，分别表示格式化之后的数字字符串，整数部分，小数部分
+ * @description: 格式化数字函数，用于将数字以及数字字符串按精度格式成对应的数字字符串，
+ * 如果用户输入是非合法的数字或数字字符串
+ * 输入项将直接作为value属性的值，同时 int 和 decimal 的值将都为空字符串
+ */
+export interface FormatedNum {
+  value: string
+  int: string
+  decimal: string
+}
+
+export function formatNum(value: string | number, precision: number): FormatedNum {
+  // 进行小数位精度补充，需要考虑value值为以下情况的场景：
+  // 1. 合法的数字或数字字符串: 100 or 100.1
+  // 2. 不合法的数字或数字字符串： 11.11.22
+  // 3. 普通文本： hello or h.i
+
+  const decimalSeparator = '.'
+  const numReg = /^(\d+)(\.(\d+))?$/
+  const numMatchRet = String(value).match(numReg)
+
+  // 不是合法的数字或数字字符串的无视精度配置项，直接返回文本作为整数部分，小数部分不展示
+  if (!numMatchRet) {
+    return {
+      value: String(value),
+      int: '',
+      decimal: '',
+    }
+  } else {
+    const decimal = String(numMatchRet[3] || '')
+      .slice(0, precision)
+      .padEnd(precision, '0')
+
+    return {
+      value: decimal.length > 0 ? `${numMatchRet[1]}${decimalSeparator}${decimal}` : numMatchRet[1],
+      int: String(numMatchRet[1]),
+      decimal: decimal,
+    }
+  }
+}
```

Comment:
```suggestion
export const numberFormatter: NumberFormatter = (value: string | number, precision: number) => {
  const numReg = /^(\d+)(\.(\d+))?$/
  const numMatchRet = String(value).match(numReg)

  if (!numMatchRet) {
    return { value: String(value) }
  } else {
    const [int, separator, _decimal = ''] = numMatchRet
    let decimal = _decimal.slice(0, precision).padEnd(precision, '0')
    decimal =  decimal.length > 0 ? `${separator}${decimal}` : ''

    return {
      value: decimal.length > 0 ? `${int}${decimal}` : int,
      int,
      decimal,
    }
  }
}
```

这个文件的名字也改成 numberFormatter 吧